script.rb

```diff
@@ -0,0 +1,228 @@
+# rubocop: disable Metrics/PerceivedComplexity, Metrics/CyclomaticComplexity, Metrics/ModuleLength, Metrics/MethodLength
+
+module Enumerable
+  def my_each
+    return to_enum unless block_given?
+
+    elements = to_a
+    elements.each do |element|
+      yield element
+    end
+    if self.class == Range
+      self
+    else
+      elements
+    end
+  end
+
+  def my_each_with_index
+    return to_enum unless block_given?
+
+    elements = to_a
+    i = 0
+    elements.each do |element|
+      yield element, i
+      i += 1
+    end
+    if self.class == Range
+      self
+    else
+      elements
+    end
+  end
+
+  def my_select
+    return to_enum unless block_given?
+
+    elements = to_a
+    result = []
+    elements.my_each do |element|
+      result.push(element) if yield element
+    end
+    result
+  end
+
+  def my_all?(arg = nil)
+    elements = to_a
+    if arg == Numeric
+      elements.my_each do |element|
+        return false unless element.class == Integer || element.class == Float
+      end
+      true
+    elsif arg.class == Class
+      elements.my_each do |element|
+        return false unless element.class == arg
+      end
+      true
+    elsif arg.class == Regexp
+      elements.my_each do |element|
+        return false unless element =~ arg
+      end
+      true
+    elsif block_given?
+      elements.my_each do |element|
+        return false unless yield element
+      end
+      true
+    elsif !block_given? && !arg
+      elements.my_each do |element|
+        return false unless element
+      end
+      true
+    elsif arg.class != Regexp && arg.class != Class
+      elements.my_each do |element|
+        return false unless element == arg
+      end
+      true
+    end
+  end
```

Comment:
- [ ] When a class is passed as an argument, `my_all?` should return true if all of the members of the collection is a member of the class (Numeric):
```ruby
 [1, 2i, 3.14].my_all?(Numeric) === [1, 2i, 3.14].all?(Numeric) => false
```